request_test.go

```diff
@@ -0,0 +1,108 @@
+package profitbricks
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+const requestTestDataPath = "testdata"
+
+type SuiteRequest struct {
+	suite.Suite
+	client *Client
+	apiUrl string
+}
+
+func (s *SuiteRequest) SetupTest() {
+	s.client = NewClient("", "")
+	s.apiUrl = s.client.client.apiURL
+	httpmock.Activate()
+}
+
+func (s *SuiteRequest) TearDownTest() {
+	httpmock.DeactivateAndReset()
+}
+
+type SuiteWaitTillRequests struct {
+	SuiteRequest
+}
+
+func Test_WaitTillRequests(t *testing.T) {
+	suite.Run(t, new(SuiteWaitTillRequests))
+}
+
+func loadTestData(t *testing.T, filename string) []byte {
+	data, err := ioutil.ReadFile(filepath.Join(requestTestDataPath, filename))
+	assert.NoError(t, err)
+	return data
+}
+
+func (s *SuiteWaitTillRequests) Test_OK_NoSelector() {
+	listResponses := [][]byte{
+		loadTestData(s.T(), "request_request_till_no_request_matches_01.json"),
```

Comment:
what's the doubled "request" in the file name's beginning for? instead of the "01" and "02" suffixes using "queued" and "done" seems better, analogous to the other 2 fixture files

Reply:
reusability. I thought that request_queued and request_done could certainly be used in other tests. While a list with a mixed set of queued and done requests, or only finished requests seemed to be to special. The name is a result of request + test